apis/hive/v1/clusterpool_types.go

```diff
@@ -103,6 +103,14 @@ type ClusterPoolStatus struct {
 	// Ready is the number of unclaimed clusters that have been installed and are ready to be claimed.
 	Ready int32 `json:"ready"`
 
+	// PoolVersion is an opaque value representing the state of the important (to
+	// ClusterDeployments) fields of the ClusterPool. When creating a new ClusterDeployment,
+	// this value is copied into its Spec.ClusterPoolReference.PoolVersion. Comparing these values
+	// gives a quick indication as to whether the ClusterDeployment's configuration matches that of
+	// the pool.
+	// +optional
+	PoolVersion string `json:"poolVersion,omitempty"`
```

Comment:
Version to me has a lot of baggage in terms of the format, an opaque digest doesn't fit well. Can we use some other name here?
🤔

Reply:
I was following the naming of `ResourceVersion`, which is a close corollary to this field: an opaque string that you're only supposed to compare for (in)equality to see if something has changed. Open to other suggestions.

Reply:
I named the annotation `cluster-pool-spec-hash`. There are still vestiges of `poolVersion` in comments, variable names, method names. Hope that's acceptable. (I really didn't want to be naming variables `clusterPoolSpecHash`...)